PokeDemo/Controllers/PokemonListViewController.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  ViewController.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 05/07/22.
+//
+// MARK: - Navigation
+
+import UIKit
+
+final class PokemonListViewController: BaseViewController {
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet private weak var tableView: UITableView! {
+        didSet {
+            tableView.register(UINib(nibName: PokemonListCell.reuseIdentifier, bundle: nil), forCellReuseIdentifier: PokemonListCell.reuseIdentifier)
+            tableView.dataSource = self
+        }
+    }
+    
+    // MARK: - Properties
+    
+    // MARK: - Private
+    
+    private let presenter = PokemonListPresenter()
+    private var pokemons: [Pokemon]? = [Pokemon]()
+    
+    // MARK: - View Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        presenter.setViewDelegate(delegate: self)
+        presenter.viewDidLoad()
+    }
+}
+
+// MARK: - Extensions
+
+extension PokemonListViewController: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return pokemons?.count ?? 0
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: PokemonListCell.reuseIdentifier) as? PokemonListCell else { return UITableViewCell() }
+        guard let pokemons else { return PokemonListCell() }
+        if indexPath.row <= pokemons.count {
+            cell.setupData(data: pokemons[indexPath.row])
+            presenter.didSetupCellWith(pokemon: pokemons[indexPath.row]) { image in
+                if let image = image{
+                    DispatchQueue.main.async {
+                        cell.setupImage(image: image)
+                    }
+                }
+            }
+        }
+        return cell
+    }
```

Comment:
could be much cleaner

```
guard let cell = tableView.dequeueReusableCell(withIdentifier: PokemonListCell.reuseIdentifier) as? PokemonListCell,
guard indexPath.row <= pokemons.count else { UITableViewCell() }
cell.setupData(data: pokemons[indexPath.row])
presenter.didSetupCellWith(pokemon: pokemons[indexPath.row]) { image in
    if let image = image {
        DispatchQueue.main.async {
            cell.setupImage(image: image)
        }
    }
}
return cell
```

Reply:
Covered in 7416c016974e6b9ad628b5d993d1f038512fa7a9